drivers/metal/rendering_device_driver_metal.mm

```diff
@@ -1058,7 +1058,7 @@ static const API_AVAILABLE(macos(11.0), ios(14.0)) MTLSamplerBorderColor SAMPLER
 
 #pragma mark - Shader
 
-const uint32_t SHADER_BINARY_VERSION = 3;
+const uint32_t SHADER_BINARY_VERSION = 4;
```

Comment:
We have to include a flag to indicate whether the shader was compiled to generate an ABI with argument buffers or classic binding model.